### ISSUE TITLE
Khansela, KULUNGILE

### ISSUE THREAD

**Issue by grinapo** [guardianproject/ChatSecureAndroid#685]
I'm not kidding. :-) On android client when I tried to change XMPP Resource (or technically really anything) the buttons say that in whatever language. Scary. I believe KULUNGILE sounds less scary, and it works. I still prefer "OK".

**Comment by tosilamaj** [comment]
why are the library folders in the external folder all empty, my android studio is doing some real complaining , pls help me

**Comment by srguglielmo** [comment]
@grinapo I got this problem too. The solution is to change ChatSecure to another language (eg, Spanish), then change it back to English. The dialog boxes now say "Ok" and "Cancel"

**Comment by alindt** [comment]
Nope. I tried to switch languages and not only it didn't fix the problem, now I have 3 languages scattered all over the place: English (the majority of the UI), French (notifications) and what appears to be Zulu ("Khansela/Kulungile") for confirmations. It's funny how Zulu isn't even listed in Settings > Languages.

![screenshot_1](https://cloud.githubusercontent.com/assets/1379385/9050686/6cf8e27e-3a59-11e5-96bc-22457a13c80e.png)
![screenshot_2](https://cloud.githubusercontent.com/assets/1379385/9050687/6d04323c-3a59-11e5-822a-469d0144b5f8.png)

**Comment by 459below** [comment]
At least I'm not the only one bothered by this. I am looking into ChatSecureAndroid/src/info/guardianproject/util/Languages.java. Seems promising that one would find something here.

For the record: I think it's rather silly that this option exists at all. Who would even want to change the language of one single app on the phone separately? Maybe there's some reason I fail to see, but I think every app should always honour the language settings of the android system.

**Comment by 459below** [comment]
I took a peek into the most recent developer branch of ChatSecure or Zom. Since there seems to be much development activity for the next version these days, I'm guessing that it will be fixed by then.

**Comment by phoenix-nz** [comment]
@459below - As someone who has been raised on more than one language I have to strongly disagree with you.
There is little more annoying than not being able to change the application language to something different than the system language.

For example almost all navigation apps sound a lot better in the native language of the country I am driving in. (As street names are almost undecipherable in english while driving in France or Germany)

Also some local apps are very badly translated. I prefer using the 'native' language of the app in that case, instead of some bad translation into my system language.

So please drop the assumption that everyone wants to use only one single language on their phone at any one time.

**Comment by n8fr8** [comment]
We definitely won't remove the ability to manually set the language. Otherwise, we've found the problem and fixes this in our some our work on the major v15 change, and are backporting the fix for this now for a v14.x.x update this week.

**Comment by grinapo** [comment]
Kulungile. ;-)

**Comment by 459below** [comment]
If the fix meant is https://github.com/guardianproject/ChatSecureAndroid/commit/1990815d0b2c98ab6c5d1a019170263ee24f6a01 (if not nevermind the following) it didn't fix the issue for my build of 14.2.1 and on a API level 19 device it behaves as such:

-rebooted device (language has not been explicitly set)
-go into chat secure 
-go to settings
-click on something with dialog -> dialog renders correctly
-back out to chat list
-go back to settings
-click on something with dialog -> dialog renders in zulu

And I really don't want to deviate any further from the issue, but I was stating my opinion and I was not suggesting removal.

**Comment by n8fr8** [comment]
Thanks for the testing, @459below. I will make sure to test on API 19 device and try to repro with the latest code. The fix was meant to make our code match other implementations of our in-app locale switching, where we haven't seen this problem. Otherwise, it seems more like a bug now in the preference rendering provided by Android, since we don't actually manually build the dialog boxes where the error is being displayed.

**Comment by 459below** [comment]
It seems in my case this error for 14.2.1 was related to the running custom rom. I couldn't reproduce this error on any other device/emulator I tested. However I could reproduce it every time on 4.4/5.1 Omnirom/CyanogenMod roms. Using a 4.3 stock rom this error was finally not reproducible on the device in question. I'm sorry for not including that information. I did not deem it to be a possible source.

**Comment by pixelrust** [comment]
I can reproduce this in 14.2.2 (Android 4.4.4 / CyanogenMod) by tapping "Activate KeySync" (no need to actually use the feature) and apparently other settings.

@srguglielmo 's workaround seems to work until you change settings again.

Anyway if @n8fr8 's hypothesis (i.e. it's an Android bug) is still valid I guess this issue may be closed.

**Comment by collectorgeneral** [comment]
For the record, this was present in settings ->Background Image, or Settings ->Heartbeat interval

When I clicked on the language option, I noticed that no option was selected by default (none of the radio buttons were 'full'). When I selected English, this issue went away. I think the issue is that the default language is not properly assigned.
